src/couch/src/couch_att.erl

```diff
@@ -510,19 +517,23 @@ flush_data(Fd, Fun, Att) when is_function(Fun) ->
                     % WriterFun({0, _Footers}, State)
                     % Called with Length == 0 on the last time.
                     % WriterFun returns NewState.
-                    fun({0, Footers}, _) ->
+                    fun({0, Footers}, _Total) ->
```

Comment:
but why bind a var (like in allocate memory) if you don't need it?

Reply:
Apparently it's a free optimization in recent code. 

http://erlang.org/documentation/doc-5.8/doc/efficiency_guide/myths.html (see the last one)

And it was confusing to read that code before so added it for clarity.

Reply:
There are a comment describing writer's fun like 3 lines above...

Sure, setting var is not slowing things down, but you still bind the var
```
1> _A = a.
a
2> _A = b.
** exception error: no match of right hand side value b
```

Personally I think this type of pseudo-documentation is an anti-pattern, it's like leaving a land mine for someone setting same `_Var` in same context. I usually don't feel too strong about it to complain in fresh code, but here you are explicitly changing existing things for not very convincing reason.

Reply:
In this case the meaning of `State` was changed from being a "don't care" 'ok' to something else that's actually used. That's the reason to change it.